swarm/api/manifest.go

```diff
@@ -578,9 +590,284 @@ func RegularSlashes(path string) (res string) {
 }
 
 func (mt *manifestTrie) getEntry(spath string) (entry *manifestTrieEntry, fullpath string) {
+	log.Debug("api.manifest.getEntry", "spath", spath)
 	path := RegularSlashes(spath)
 	var pos int
 	quitC := make(chan bool)
 	entry, pos = mt.findPrefixOf(path, quitC)
 	return entry, path[:pos]
 }
+
+func (a *API) UpdateManifest(ctx context.Context, addr storage.Address, update func(mw *ManifestWriter) error) (storage.Address, error) {
```

Comment:
why are these moved?

Reply:
ideally we should have 1 type `Manifest` that you can perform an `Update` action on. right now the code that does manifest modifications is spread out on different files, contexts and types. it should all be governed from one place.